src/store/items.ts

```diff
@@ -8,14 +8,21 @@ import merge from 'lodash/merge';
 import some from 'lodash/some';
 import union from 'lodash/union';
 
+interface ItemsByType {
+  Type: string;
+  Items: BaseItemDto[];
+}
+
 export interface ItemsState {
   byId: Record<string, BaseItemDto>;
   allIds: readonly string[];
+  boxSetChildrenById: Record<string, ItemsByType[]>;
 }
 
 export const defaultState = (): ItemsState => ({
   byId: {},
-  allIds: []
+  allIds: [],
+  boxSetChildrenById: {}
```

Comment:
Why we're treating the collection items differently? They're normal items, just with a parent.

Reply:
I have added the fixes you pointed out to me above, 
however I have a problem with this one.


I was trying to modify the items store to add only child items, calling
```ts
response.Items? .forEach ((item) => {
      commit ('ADD_ITEM', {item});
    });
    commit ('ADD_ITEMS', {
      items: response.Items
    });
```
from the Success function of my action, to avoid having the 'boxSetChildrenById' property.


I modified my getter to return the appropriate data by extracting it from the state.byId but I found that the **parentId** of the elements I get by **filtering by parentId in the api** call, does _**not match**_ that of the collection (because in fact I believe it matches the reference library instead , as is correct).

**For these reasons**, I am **not** being able to recover the elements that are children of the collection from my getter (from state.byId), which is why I had opted to save them in this way.

This being the case (to avoid keeping duplicates in the store) it might be better to keep a similar property, but to save only the references (id) for collections to the state.byId?

What do you think? Suggestions?

Reply:
Fair enough and thanks for the finding. Let's leave it like this, code evolves and always can be revised later :)

But for now, let's not overengineer stuff.

Reply:
Perfect! I leave it like this and throw local changes
Thanks for taking the time!

I am happy to help the project and I intend to continue doing it as much as I can when time permits!

I like the project and it is the first github project in which I collaborate :)

Ps. I noticed a test failed, but only for a failed call to some github service

Reply:
In relation to this (3 comments back)
> I modified my getter to return the appropriate data by extracting it from the state.byId but I found that the **parentId** of the elements I get by **filtering by parentId in the api** call, does _**not match**_ that of the collection (because in fact I believe it matches the reference library instead , as is correct).
>

it would not be convenient to keep in mind this need to get the children elements of the collections regarding #801?

They seem to require proper handling as indicated in the diagram regarding _ShowSeasons_, don't you think?

Reply:
I don't understand what you mean, care to make an example?

Don't pay much attention to that diagram either, it's not final and we've not set ourselves so much in the need of some stores (like for the home sections).

Reply:
I mean that given the impossibility of obtaining the children of a particular element (except for the original folder) from the generic list of elements in the store, I believe the need arises to have an appropriate management of the collections (since having a "boxSetChildrenByType" property does not it is properly the competence of the item store, as you pointed out to me)

But it's probably best to bring the topic to the relative issue we were talking about

Reply:
@doc-code-hub Oh yes, so you mean leaving it exactly how it's right now in this PR, correct? For me the current implementation is fine for the time being, until we come up with something better with the whole refactor and everything it's redone.